cpp/src/arrow/python/helpers.cc

```diff
@@ -373,6 +374,8 @@ Status IntegerScalarToFloat32Safe(PyObject* obj, float* out) {
   return Status::OK();
 }
 
+void DebugPrint(PyObject* obj) { std::cerr << PyObject_StdStringRepr(obj) << std::endl; }
```

Comment:
For the record, I would generally use [PySys_FormatStderr](https://docs.python.org/3/c-api/sys.html#c.PySys_FormatStderr).

Reply:
done